wcfsetup/install/files/js/WoltLabSuite/Core/Clipboard.js

```diff
@@ -0,0 +1,34 @@
+/**
+ * Wrapper around the web browser's various clipboard APIs.
+ * 
+ * @author	Tim Duesterhus
+ * @copyright	2001-2018 WoltLab GmbH
+ * @license	GNU Lesser General Public License <http://opensource.org/licenses/lgpl-license.php>
+ * @module	WoltLabSuite/Core/Clipboard
+ */
+define([], function() {
+	"use strict";
+	
+	return {
+		copyElementTextToClipboard: function (element) {
+			var promise;
+			if (navigator.clipboard) {
+				promise = navigator.clipboard.writeText(element.innerText);
+			}
+			else {
+				promise = Promise.reject('navigator.clipboard is not supported');
+			}
+			
+			return promise.catch(function () {
+				if (!window.getSelection) throw new Error('window.getSelection is not supported');
+				
+				var range = document.createRange();
+				range.selectNode(element);
+				window.getSelection().addRange(range);
+				if (!document.execCommand('copy')) {
```

Comment:
`document.execCommand('copy')` fails in Edge, probably because it is being called in a Promise and not synchronously in the event handler.

Reply:
Included in 8d1094cc519f0056c80f24613ac1596c56921e22, but I was not able to verify the fix due to the lack of Edge. Internet Explorer 11 works fine.